### PR TITLE
feat(builder): UX and style updates

### PR DESCRIPTION
**Pull request by kcze**
### Changes 🏗️

Screenshot 2024-07-22 at 4 46 51 PM

- Handles:
  - Add `NodeHandle` to draw input and output handles
  - Position handles relatively
  - Make entire handle label clickable/connectable
  - Add input/output types below labels
  - Change color on hover and when connected
  - "Connected" no longer shows up when connected
- Edges:
  - Draw edge above node when connecting to the same node
  - Add custom `ConnectionLine`; drawn when making a connection
  - Add `CustomEdge`; drawn for existing connections
  - Add arrow to the edge end
  - Colorize depending on type
- Input field modal:
  - Select all text when opened
  - Disable node dragging
- CSS:
  - Remove not needed styling
  - Use tailwind classes instead of css for some components
  - Minor style changes
- Add shadcn switch
- Change bottom node buttons (for properties and advanced) to switches
- Format code

Known bugs:
- Colorize doesn't work when loading flow
- Colorize sometimes doesn't work for self-connected edges
- Edges start and end position depend on handle width so looks bad when handle label is too short/long
- If there's no type, edge is drawn incorrectly
- When dragging connection, the snapped line end is at the center of handle, not on the connector dot 

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Review by Swiftyos — DISMISSED** [review]

**Comment by Pwuts** [comment]
image
Can this be a switch too?

File: rnd/autogpt_builder/src/components/customnode.css [hunk 1]
```diff
@@ -206,13 +172,6 @@
...
-  height: 10px;
-}
-
-/* Make edges thicker */
-.react-flow__edge {
-  stroke-width: 2px !important;
-}
+.custom-switch {
+  padding-left: 2px;
+}
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/components/customnode.css]: Missing newline

File: rnd/autogpt_builder/src/components/CustomEdge.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <BaseEdge
+      style={{ strokeWidth: 2, stroke: (data?.edgeColor ?? '#555555') + (selected ? '' : '80') }}
+      path={edgePath}
+      markerEnd={markerEnd}
+    />
+  )
+};
+
+
+export const CustomEdge = memo(CustomEdgeFC);
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/components/CustomEdge.tsx]: Missing newline

**Review by Pwuts — APPROVED** [review]
Very nice!

**Review by aarushik93 — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

**Pull request merged**